sycl/doc/EnvironmentVariables.md

```diff
@@ -14,6 +14,7 @@ subject to change. Do not rely on these variables in production code.
 | SYCL_PI_TRACE | Described [below](#sycl_pi_trace-options)  | Enable specified level of tracing for PI. |
 | SYCL_BE | PI_OPENCL, PI_LEVEL_ZERO, PI_CUDA | Force SYCL RT to consider only devices of the specified backend during the device selection. |
```

Comment:
Can we put a note that SYCL_BE and SYCL_DEVICE_TYPE are deprecated, and one should use SYCL_DEVICE_FILTER instead?

Reply:
We are not ready for deprecation yet because this can affect customers usage model a lot.
The deprecation process requires many discussions and approvals with QA, customer relations, and documentation team.
I will start discussion once this PR is merged.

Reply:
Yes, we introduce this new env var so customers can start using it.
We need to give some grace period to customers before we deprecate the old ones.

Reply:
Can we add a warning if "deprecated" variables are set?

Reply:
I was told not to mention "deprecation" until we fully discuss and make a deprecation plan with the other teams such as QA, customer relation, docs team.

Reply:
and name the SYCL_DEVICE_FILTER in the warning and in this doc of SYCL_BE and SYCL_DEVICE_TYPE

Reply:
I agree.
We will need a follow up deprecation plan and at that time, I will add the deprecation notice in the code so the users can notice the warning during the grace period.

Reply:
Great. Thanks!